panel/io/location.py

```diff
@@ -164,9 +182,10 @@ def _update_synced(self, event: param.parameterized.Event = None) -> None:
                 except Exception:
                     pass
                 try:
-                    equal = v == getattr(p, pname)
+                    equal = _is_equal(v, getattr(p, pname), type(p.param[pname]))
```

Comment:
This probably should just be something like:

```
v is (new:= getattr(p, pname)) or _generate_hash(v) == _generate_hash(new)
```

where `_generate_hash` is taken from `panel.io.cache`. Or rather we should add a `is_equal` that does that to `pn.io.cache` and then import it here.